### ISSUE TITLE
Reuse: 5 tiny helpers (CPF normalize + filename date + day-of-week + slug + array sanitize)

### ISSUE THREAD

**Issue by rpgmem** [rpgmem/ffcertificate#275]
Sub-escopo extraído de #254 §4 — consolidar 5 padrões duplicados pequenos numa única PR com helpers tipados.

> **Workflow:** plano fechado abaixo; 3 sprints atômicos numa PR única; sem version-bump intermediário — tudo consolida em **v6.6.1**.

---

## Regra 6 — counts ajustados

| Pattern | #254 dizia | Real | Helper |
|---|---|---|---|
| CPF normalize (`preg_replace('/[^0-9]/','',...)`) | 4 | **10** | `DataSanitizer::normalize_cpf_rf(string $value): string` |
| Filename date em export CSV | 6 | 6 | `Utils::get_export_filename(string $prefix, ?string $title = null): string` |
| Day-of-week `(int) gmdate('w', $ts ?: time())` | 5 | 5 | `Utils::get_day_of_week_number(?int $ts = null): int` |
| Username slug 2-step regex | 6 | **2** (override threshold pra padronizar) | `Utils::sanitize_username_slug(string $value): string` |
| `array_map('sanitize_text_field', wp_unslash($_POST[...]))` | 11 | **4** | `Utils::get_post_array(string $key, array $default = []): array` |

**Total: 5 helpers + ~27 migrations.**

## Localizações

**CPF normalize (10):**
- `class-ffc-admin-ajax.php:232`
- `class-ffc-submission-handler.php:143`
- `class-ffc-self-scheduling-appointment-handler.php:484`
- `class-ffc-self-scheduling-appointment-validator.php:246`
- `class-ffc-reprint-detector.php:73`
- `class-ffc-form-processor.php:289`, `:685`
- `ffc-appointment-repository.php:155`, `:517`
- `ffc-submission-repository.php:196`

**Filename date (6):**
- `class-ffc-csv-exporter.php:134`
- `class-ffc-admin-activity-log-page.php:168`
- `class-ffc-reregistration-csv-exporter.php:58`
- `class-ffc-self-scheduling-appointment-csv-exporter.php:259`
- `class-ffc-audience-admin-import.php:423`, `:476`

**Day-of-week (5):**
- `class-ffc-working-hours-service.php:51`, `:101`, `:136`
- `class-ffc-self-scheduling-appointment-handler.php:250`
- `ffc-blocked-date-repository.php:292`

**Username slug (2):**
- `class-ffc-user-creator.php:273-277` (pattern: `sanitize_user` + `remove_accents` + 2 regex + trim)
- `class-ffc-user-creator.php:304-308` (mesmo pattern)

**array_map sanitize (4):**
- `class-ffc-settings-save-handler.php:79`
- `class-ffc-settings-save-handler.php:390`
- `class-ffc-tab-user-access.php:97`
- `class-ffc-ajax-trait.php:142`

## Decisões de helper home

| Helper | Home |
|---|---|
| `normalize_cpf_rf` | `Core\DataSanitizer` (sanitize-domain) |
| `get_export_filename` | `Core\Utils` (general utility) |
| `get_day_of_week_number` | `Core\Utils` (general utility) |
| `sanitize_username_slug` | `Core\Utils` (general utility — não vale criar `StringNormalizer` só pra 1 helper) |
| `get_post_array` | `Core\Utils` (request-input utility, par com futuro `get_post_string`) |

---

## Plano — 3 sprints (1 commit cada)

Branch sugerida: **`claude/reuse-tiny-helpers`**.

### Sprint 1 — Criar 5 helpers + tests

**Arquivos:**
- `includes/core/class-ffc-data-sanitizer.php` (existente) — adicionar `normalize_cpf_rf`.
- `includes/core/class-ffc-utils.php` (existente) — adicionar `get_export_filename`, `get_day_of_week_number`, `sanitize_username_slug`, `get_post_array`.
- `tests/Unit/DataSanitizerTest.php` (existente) — cases para `normalize_cpf_rf`.
- `tests/Unit/UtilsTest.php` (existente) — cases para os 4 utils.

**Helpers (signatures + behaviors):**

```php
// DataSanitizer
public static function normalize_cpf_rf( string $value ): string {
    return preg_replace( '/[^0-9]/', '', $value ) ?? '';
}

// Utils
public static function get_export_filename( string $prefix, ?string $title = null ): string {
    $parts = array( $prefix );
    if ( null !== $title && '' !== $title ) {
        $parts[] = sanitize_file_name( $title );
    }
    $parts[] = gmdate( 'Y-m-d' );
    return implode( '-', $parts ) . '.csv';
}

public static function get_day_of_week_number( ?int $ts = null ): int {
    return (int) gmdate( 'w', $ts ?? time() );
}

public static function sanitize_username_slug( string $value ): string {
    $slug = sanitize_user( remove_accents( $value ), true );
    $slug = preg_replace( '/[^a-z0-9._-]/', '', $slug ) ?? '';
    $slug = preg_replace( '/[-_.]+/', '.', $slug ) ?? '';
    return trim( $slug, '.' );
}

public static function get_post_array( string $key, array $default = array() ): array {
    if ( ! isset( $_POST[ $key ] ) ) { return $default; }
    // phpcs:ignore WordPress.Security.NonceVerification.Missing -- Caller responsibility.
    $raw = wp_unslash( $_POST[ $key ] );
    if ( ! is_array( $raw ) ) { return $default; }
    return array_map( 'sanitize_text_field', $raw );
}
```

**Test coverage (~25-30 new cases)**: empty/null inputs, edge cases (CPF with dots/dashes, filename with/without title, day boundaries 0-6, slug with accents, post_array with non-array input).

**Commit:** `Add 5 reuse helpers in DataSanitizer + Utils (refs #254, #275)`

### Sprint 2 — Migrar ~27 call sites

Migrações mecânicas por pattern:

| Pattern atual | Substituir por |
|---|---|
| `preg_replace('/[^0-9]/', '', $cpf_rf) ?? ''` | `DataSanitizer::normalize_cpf_rf( $cpf_rf )` |
| `(string) preg_replace('/[^0-9]/', '', $val)` | `DataSanitizer::normalize_cpf_rf( $val )` |
| `'<prefix>-' . gmdate('Y-m-d') . '.csv'` | `Utils::get_export_filename( '<prefix>' )` |
| `'<prefix>-' . sanitize_file_name($title) . '-' . gmdate('Y-m-d') . '.csv'` | `Utils::get_export_filename( '<prefix>', $title )` |
| `(int) gmdate('w', $date_ts ? $date_ts : time())` | `Utils::get_day_of_week_number( $date_ts ?: null )` |
| Username slug 4-line block | `Utils::sanitize_username_slug( $value )` |
| `isset(...) ? array_map('sanitize_text_field', wp_unslash($_POST[...])) : array()` | `Utils::get_post_array( '<key>' )` |

**Cuidado em Sprint 2:**
- Alguns CPF normalize têm pré-processings inline (e.g. cast to string before regex). Helper já cobre.
- Some get_export_filename callers prefix with `\FreeFormCertificate\Core\Utils::sanitize_filename($title)` first — verificar se substitui equivalente ao `sanitize_file_name`.

**Commit:** `Migrate 27 call sites to reuse helpers (refs #254, #275)`

### Sprint 3 — CHANGELOG

```markdown
### Changed

- 5 new reuse helpers in `Core\DataSanitizer` and `Core\Utils` consolidate
  scattered patterns: `normalize_cpf_rf` (10 sites), `get_export_filename`
  (6 sites, CSV download), `get_day_of_week_number` (5 sites, scheduling),
  `sanitize_username_slug` (2 sites, user-creator), `get_post_array`
  (4 sites, settings save handlers). 27 call sites migrated.
```

**Commit:** `Add CHANGELOG entry for #275 (refs #254, #275)`

---

## Fluxo de PR

1. Criar branch `claude/reuse-tiny-helpers` a partir de `main`.
2. Executar Sprints 1→3.
3. `mcp__github__create_pull_request` draft → não-draft → auto-merge SQUASH.

---

## Critério de pronto

- [ ] 5 helpers existem com tests (~25-30 cases).
- [ ] 27 call sites migrados.
- [ ] `grep -rn "preg_replace('/\[\^0-9\]/'" includes/` ⇒ 0 hits.
- [ ] `grep -rn "gmdate('Y-m-d') . '.csv'" includes/` ⇒ 0 hits.
- [ ] `grep -rn "gmdate('w', \\\$" includes/` ⇒ 0 hits (ou só dentro do helper).
- [ ] PHPStan + WPCS + PHPUnit verdes.
- [ ] CHANGELOG `[Unreleased]` populado.

---

Refs #254.
